### PR TITLE
Scattering-freefield

### PR DESCRIPTION
**Pull request by ahms5**
### Changes proposed in this pull request:
- implement scattering coefficient calculation from freefield measurements/simulations

### CODE REVIEW

File: imkar/testing/stub_utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""
```
Comment (sikersten) [inline imkar/testing/stub_utils.py:1]: If you really want to provide stub utils, please add test for them

File: imkar/scattering/coefficient.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    if not isinstance(p_reference, pf.FrequencyData):
+        raise ValueError("p_reference has to be FrequencyData")
+    if not isinstance(mics, pf.Coordinates):
+        raise ValueError("mics have to be Coordinates")
+    if (incident_directions is not None) & \
+            ~isinstance(incident_directions, pf.Coordinates):
+        raise ValueError("incident_directions have to be None or Coordinates")
+
+    # calculate according to mommertz correlation method
+    p_sample_sq = p_sample*p_sample
```
[the remark this answers is not in the document]
Comment (ahms5) [reply imkar/scattering/coefficient.py:56]: good point, I included them

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import pytest
+import numpy as np
+import pyfar as pf
+from imkar.testing import stub_utils
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (ahms5) [reply tests/conftest.py:6]: The shape of the data is quite different, pyfar.samplings just returns a flatten Coordinates object. we need a cshape of (#angle_1, #angle_2)

**Comment by ahms5** [comment]
ready for review, see #7 at first.  #7 is merged into this here.

**Comment by ahms5** [comment]
doc fails, fixes in #10

**Pull request closed without merging**